pkg/apis/pipeline/v1beta1/param_types.go

```diff
@@ -112,12 +112,27 @@ type ArrayOrString struct {
 
 // UnmarshalJSON implements the json.Unmarshaller interface.
 func (arrayOrString *ArrayOrString) UnmarshalJSON(value []byte) error {
-	if value[0] == '"' {
+	if len(value) == 0 {
 		arrayOrString.Type = ParamTypeString
-		return json.Unmarshal(value, &arrayOrString.StringVal)
+		arrayOrString.StringVal = string(value)
+		return nil
 	}
-	arrayOrString.Type = ParamTypeArray
-	return json.Unmarshal(value, &arrayOrString.ArrayVal)
+	switch value[0] {
+	case '[':
+		arrayOrString.Type = ParamTypeArray
+		if err := json.Unmarshal(value, &arrayOrString.ArrayVal); err == nil {
+			return nil
```

Comment:
I realized the errors didn't get thrown here. This should be `err != nil` then `return nil`

Reply:
The reason I did this is because for case like nested lists or lists using not string element, the UnmarshalJson won't work.
Like `[1,2,3]` with int elements, but I think we should Unmarshal it into a whole string. So if we encountered err when trying to unmarshal to array, we will continue and unmarshal it to a string. 

I agree this part of code is not writing very well.

Reply:
Ah I see. Then this should be fine. We will capture the type mismatch when you validate the results against PropertiesSpec. So it definitely worths adding more comments here and in `filterResultsAndResources()` to explain the details

Reply:
Sure! Thanks for the suggestions!